src/core/mac/sub_mac.cpp

```diff
@@ -447,7 +447,7 @@ void SubMac::SampleRssi(void)
 
     if (TimerMilliScheduler::IsStrictlyBefore(TimerMilli::GetNow(), mEnergyScanEndTime))
     {
-        mTimer.StartAt(mTimer.GetFireTime(), kEnergyScanRssiSampleInterval);
+        mTimer.StartAt(mTimer.GetFireTime(), static_cast<int32_t>(kEnergyScanRssiSampleInterval));
```

Comment:
Do (some) compilers need the explicit type conversion (`static_cast<int32_t>`)? (won't treat the constant as `int32_t`)?

Reply:
Yes, some compiler will generate warnings about ambiguity. In fact I think we might define non-enumeration constants with `static const`. What do you think?